### PR TITLE
Capture response headers and redact all captured headers

### PR DESCRIPTION
**Pull request by tsiq-karold**
Closes https://github.com/twosigma/webtau/issues/354.

### CODE REVIEW

File: webtau-http-groovy/src/test/groovy/com/twosigma/webtau/http/HttpGroovyTest.groovy [hunk 1]
```diff
@@ -468,6 +471,72 @@ class HttpGroovyTest implements HttpConfiguration {
...
+
+        Path requestBodyFile = docRoot.resolve("request.json")
+        assertTrue(Files.exists(requestBodyFile))
+        def capturedRequestBody = JsonUtils.deserializeAsMap(FileUtils.fileTextContent(requestBodyFile))
+        capturedRequestBody.should == requestBody
+
+        Path responseBodyFile = docRoot.resolve("response.json")
+        assertTrue(Files.exists(responseBodyFile))
+        def capturedResponseBody = JsonUtils.deserializeAsMap(FileUtils.fileTextContent(responseBodyFile))
+        capturedResponseBody.should == requestBody
```
Comment (MykolaGolubyev) [inline webtau-http-groovy/src/test/groovy/com/twosigma/webtau/http/HttpGroovyTest.groovy]: seems like a bit of duplication

File: webtau-http-groovy/src/test/groovy/com/twosigma/webtau/http/HttpGroovyTest.groovy [hunk 2]
```diff
@@ -468,6 +471,72 @@ class HttpGroovyTest implements HttpConfiguration {
         assertFalse(Files.exists(requestFile))
     }
 
+    @Test
+    void "appropriate capture files are generated"() {
```
Comment (MykolaGolubyev) [inline webtau-http-groovy/src/test/groovy/com/twosigma/webtau/http/HttpGroovyTest.groovy]: should we be explicit that it is about `doc` related capture?

File: webtau-http-groovy/src/test/groovy/com/twosigma/webtau/http/HttpGroovyTest.groovy [hunk 3]
```diff
@@ -468,6 +471,72 @@ class HttpGroovyTest implements HttpConfiguration {
...
+
+        Path requestHeaderFile = docRoot.resolve("request.header.txt")
+        assertTrue(Files.exists(requestHeaderFile))
+        def authorizationHeader = Files.lines(requestHeaderFile).filter { line -> line.toLowerCase().startsWith('authorization') }.findFirst().get()
+        authorizationHeader.toLowerCase().should == 'authorization: ................'
+
+        Path responseHeaderFile = docRoot.resolve("response.header.txt")
+        assertTrue(Files.exists(responseHeaderFile))
+        authorizationHeader = Files.lines(responseHeaderFile).filter { line -> line.toLowerCase().startsWith('authorization') }.findFirst().get()
+        authorizationHeader.toLowerCase().should == 'authorization: ................'
```
Comment (MykolaGolubyev) [inline webtau-http-groovy/src/test/groovy/com/twosigma/webtau/http/HttpGroovyTest.groovy]: a bit of duplication

File: webtau-http/src/main/java/com/twosigma/webtau/http/HttpDocumentation.java [hunk 4]
```diff
@@ -44,19 +44,30 @@ public Capture(String artifactName) {
...
             captureRequestBody();
             captureResponseBody();
+
             capturePaths();
         }
 
         private void captureRequestHeader() {
-            String requestHeader = lastValidationResult.getRequestHeader().toString();
-            if (! requestHeader.isEmpty()) {
+            String requestHeader = lastValidationResult.getRequestHeader().redactSecrets().toString();
```
Comment (MykolaGolubyev) [inline webtau-http/src/main/java/com/twosigma/webtau/http/HttpDocumentation.java]: nice catch

File: webtau-http-groovy/src/test/groovy/com/twosigma/webtau/http/HttpGroovyTest.groovy (hunk 1 above)
Comment (tsiq-karold) [reply webtau-http-groovy/src/test/groovy/com/twosigma/webtau/http/HttpGroovyTest.groovy]: Yeah, there's enough duplication to make it look duplicative but not enough that things look much simpler with it removed.  I've pushed a change to reduce it a bit, let me know what you think?

File: webtau-http-groovy/src/test/groovy/com/twosigma/webtau/http/HttpGroovyTest.groovy (hunk 2 above)
Comment (tsiq-karold) [reply webtau-http-groovy/src/test/groovy/com/twosigma/webtau/http/HttpGroovyTest.groovy]: Done.

**Pull request merged**